test-data/unit/check-classes.test

```diff
@@ -1942,6 +1942,17 @@ def new_pro_user(user_class: Type[ProUser]):
     new_user(user_class)
 [out]
 
+[case testAllowCovariantArgsInConstructor]
+from typing import Generic, TypeVar
+
+T_co = TypeVar('T_co', covariant=True)
+
+class C(Generic[T_co]):
+    def __init__(self, x: T_co) -> None: # This should be allowed
+       pass
```

Comment:
Also try assignment `self.x = x` in `__init__` and `reveal_type(self.x)` in a method other than `__init__`.